bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,377 @@
+#!/usr/bin/env -S python3 -u
+# Bang line above passes '-u' to python, for unbuffered output
+# Permissions required:
+#  psql - access to heroku app's postgres
+#  aws s3 - NZSL IAM access
+#  s3:GetObjectAcl permissions or READ_ACP access to the object
+#  https://docs.aws.amazon.com/cli/latest/reference/s3api/get-object-acl.html
+
+import os
+import sys
+import subprocess
+import argparse
+import re
+from pprint import pprint
+
+parser = argparse.ArgumentParser(
+    description="You must setup: An AWS auth means, eg. AWS_PROFILE env var. "
+    "Postgres access details, eg. DATABASE_URL env var."
+)
+# 'Go' mode, args.do_actions
+parser.add_argument(
+    "--do-actions",
+    action="store_true",
+    default=False,
+    required=False,
+    help="Actually perform Delete objects or change ACLs (DESTRUCTIVE operation)",
+)
+parser.add_argument(
+    "--env",
+    default="uat",
+    required=False,
+    help="Environment to run against, eg 'production, 'uat', etc (default: '%(default)s')",
+)
+parser.add_argument(
+    "--pgcli",
+    default="/usr/bin/psql",
+    required=False,
+    help=f"Postgres client path (default: %(default)s)",
+)
+parser.add_argument(
+    "--awscli",
+    default="/usr/local/bin/aws",
+    required=False,
+    help=f"AWS client path (default: %(default)s)",
+)
+parser.add_argument(
+    "--tmpdir",
+    default="/tmp/nzsl",
+    required=False,
+    help=f"Temp dir path (default: %(default)s)",
+)
+args = parser.parse_args()
+
+# Globals
+CSV_DELIMITER = ","
+DATABASE_URL = os.getenv("DATABASE_URL", "")
+AWSCLI = args.awscli
+PGCLI = args.pgcli
+AWS_S3_BUCKET = f"nzsl-signbank-media-{args.env}"
+
+
+def pg_cli(args_list):
+    if not isinstance(args_list, list):
+        args_list = [args_list]
+    return subprocess.run(
```

Comment:
nit: I don't think there's a lot of gain in supporting this, given it's our own code and it's just saving us having an extra couple of characters

```suggestion
def pg_cli(args_list):
    return subprocess.run(
```

In my experience in situations like this, this kind of feature feels like a nice optimization but really just adds more overhead with little gain; consider:
  * this function can now be used with two different types of arguments, so we have to think/decide about that every place it's used
  * if you implement another similar function, you have to choose to implement it there
    * if you don't then you're at best being inconsistent, and at worse going to have trouble later if you try to use one of those functions with the assumption it'll work, but then it doesn't...
    * if you do, then you're having to carry around code that might not be used in the other function (like it isn't for `aws_cli`), for the sake of consistency
  * ...and it just snowballs from there 😅 

to be clear, I'm not saying this kind of thing is inherently bad, just that it has a time and place, and that a custom self-contained script isn't one of them

Reply:
Done.